File: smsflow-android/app/src/main/java/com/smsflow/gateway/domain/model/Message.kt
package com.smsflow.gateway.domain.model

data class Message(
    val id: String,
    val externalId: String?,
    val phoneNumber: String,
    val body: String,
    val status: MessageStatus,
    val simSlot: Int,
    val createdAt: Long,
    val updatedAt: Long,
    val errorCode: String?
)

enum class MessageStatus {
    PENDING,
    QUEUED,
    SENDING,
    SENT,
    DELIVERED,
    FAILED;

    companion object {
        fun fromString(value: String): MessageStatus {
            return try {
                valueOf(value.uppercase())
            } catch (e: IllegalArgumentException) {
                FAILED
            }
        }
    }
}

File: smsflow-web/packages/shared/src/types/message.ts
export type MessageStatus = 'PENDING' | 'QUEUED' | 'DISPATCHED' | 'SENT' | 'DELIVERED' | 'FAILED' | 'EXPIRED';
export type MessageDirection = 'INBOUND' | 'OUTBOUND';

export interface Message {
  id: string;
  userId: string;
  deviceId?: string;
  phoneNumber: string;
  body: string;
  status: MessageStatus;
  direction: MessageDirection;
  simSlot?: number;
  templateId?: string;
  bulkBatchId?: string;
  errorCode?: string;
  errorMessage?: string;
  scheduledAt?: Date;
  sentAt?: Date;
  deliveredAt?: Date;
  createdAt: Date;
  updatedAt: Date;
}
